drift/core/psmc.py

```diff
@@ -209,6 +213,273 @@ def _work_fisher_bias_m(self, mi):
         return fisher, bias
 
 
+class PSMonteCarloXLarge(PSMonteCarlo):
+    """An extension of the PSEstimation class to support estimation of the
+    Fisher matrix via Monte-Carlo simulations for very large arrays.
+
+    Attributes
+    ----------
+    nsamples : integer
+        The number of samples to draw from each band.
+    """
+
+    nsamples = config.Property(proptype=int, default=500)
+
+    # ==== Calculate the total Fisher matrix/bias =======
+
+    def make_clzz_array(self):
+        """Store clarray for power spectrum bands in parallel with MPIArray."""
+        nbands = self.nbands
+        nfreq = self.telescope.nfreq
+        lmax = self.telescope.lmax
+
+        self.clarray = mpiarray.MPIArray(
+            (nbands, lmax + 1, nfreq, nfreq),
+            axis=0,
+            dtype=np.float64,
+            comm=MPI.COMM_WORLD,
+        )
+
+        self.clarray[:] = 0.0
+
+        for bl, bg in self.clarray.enumerate(axis=0):
+            self.clarray[bl] = self.make_clzz(self.band_pk[bg])
+
+    def generate(self, regen=False):
+        """Calculate the total Fisher matrix and bias and save to a file.
+
+        Parameters
+        ----------
+        regen : boolean, optional
+            Force regeneration if products already exist (default `False`).
+        """
+        if mpiutil.rank0:
+            st = time.time()
+            print("======== Starting PS calculation MC XLARGE ========")
+
+        ffile = self.psdir + "/fisher.hdf5"
+
+        if os.path.exists(ffile) and not regen:
+            print("Fisher matrix file: %s exists. Skipping..." % ffile)
+            return
+
+        mpiutil.barrier()
+
+        # Pre-compute all the angular power spectra for the bands.
+        self.genbands()
+
+        comm = MPI.COMM_WORLD
+        size = comm.Get_size()
+        rank = comm.Get_rank()
+
+        st = time.time()
+
+        m_chunks, low_bound, upp_bound = self.split_single(
+            (self.telescope.mmax + 1), size
+        )
+        num_chunks = m_chunks.shape[0]
+
+        if mpiutil.rank0:
+            print(
+                "M chunks, lower bounds, upper bounds", m_chunks, low_bound, upp_bound
+            )
```

Comment:
This looks like debugging info you just stuck in there but forgot to remove. If it's not, please format it properly.

Reply:
deleted.